### PR TITLE
fix: io metrics collection

### PR DESCRIPTION
**Pull request by gsanchezgavier**
- Fixes io metrics collection
- Add sampler tests

### CODE REVIEW

**Comment by gsanchezgavier** [comment]
I will stop fixing legacy linter errors , the current ones are old ones that appear because of replacing the `system.Info` struct.

File: src/biz/storage.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ type DeviceMapperStats struct {
 
 // ParseDeviceMapperStats parses the output from the info DriverStatus info Array. This information is not consider stable by
 // Docker and could change anytime according to API the docs.
-func ParseDeviceMapperStats(info types.Info) (*DeviceMapperStats, error) {
+func ParseDeviceMapperStats(info system.Info) (*DeviceMapperStats, error) {
```
Comment (marcsanmi) [inline src/biz/storage.go:24]: Why this?

Comment (gsanchezgavier) [reply src/biz/storage.go:24]: the linter failed since the type is marked as deprecated

File: src/nri/sampler.go [hunk 2]
```diff
@@ -274,16 +275,16 @@ func blkio(bio *biz.BlkIO) []entry {
 	var entries []entry
 
 	if bio.TotalReadCount != nil {
-		entries = append(entries, metricIOTotalReadCount(bio.TotalReadCount), metricIOReadCountPerSecond(bio.TotalReadCount))
+		entries = append(entries, metricIOTotalReadCount(*bio.TotalReadCount), metricIOReadCountPerSecond(*bio.TotalReadCount))
```
Comment (marcsanmi) [inline src/nri/sampler.go:278]: Good catch! We're testing the biz metrics in the `TestBlkIOMetrics` but not the sampler 👍

**Review by marcsanmi — APPROVED** [review]
LGTM! 🚀

**Pull request merged**